test/unit/session-initiation-timing.test.ts

```diff
@@ -0,0 +1,130 @@
+import { EventEmitter } from "events";
+import { Agent, createClient } from "stanza";
+import { IQ } from 'stanza/protocol';
+import { v4 as uuidv4 } from 'uuid';
+
+import { HttpClient } from "../../src/http-client";
+import { GenesysCloudMediaSession } from '../../src/types/genesys-cloud-media-session';
+import { IPendingSession } from "../../src/types/interfaces";
+import { NamedAgent } from '../../src/types/named-agent';
+import { StanzaMediaSession } from "../../src/types/stanza-media-session";
+import { WebrtcExtension } from "../../src/webrtc";
+import { SessionOpts } from "stanza/jingle/Session";
+
+jest.mock('../../src/types/stanza-media-session');
+StanzaMediaSession.prototype.on = jest.fn();
+
+jest.mock('../../src/types/genesys-cloud-media-session', () => {
+  return {
+    GenesysCloudMediaSession: jest.fn().mockImplementation((webrtcExtension, params) => ({
+      meetingId: params.meetingId,
```

Comment:
I tried to put these tests in webrtc.test.ts, but this mock caused issues with other tests, though I'm not quite sure why.

There might be another way to do this, but I wanted a way to verify that we still had a pending session for SDP over XMPP, regardless of the order we handled things in. In the context I've been working in, one way to do this is to make sure the meetingId is grabbed from the pending session and set on the session. But that happens in the constructor, which is why I had to mock this differently.

Let me know if you see another way to accomplish this.